src/app/shared/validate-password.directive.ts

```diff
@@ -0,0 +1,23 @@
+import { Directive, Input } from '@angular/core';
+import { AbstractControl, NG_VALIDATORS, Validator, ValidatorFn } from '@angular/forms';
+
+@Directive({
+    selector: '[appValidatePassword]',
+    providers: [{ provide: NG_VALIDATORS, useExisting: ValidatePasswordDirective, multi: true }],
+})
+export class ValidatePasswordDirective implements Validator {
+    @Input('appValidatePassword') validPassword: string;
+
+    constructor() {}
+
+    validate(control: AbstractControl): { [key: string]: any } | null {
+        return this.validPassword ? checkPassword(new RegExp(this.validPassword))(control) : null;
+    }
+}
+
+export function checkPassword(password: RegExp): ValidatorFn {
```

Comment:
This entire file is quite complicated. It has a validate function that returns `checkPassword(...)`. The function name sounds like it should DO something (= void function), not like it should `return` something. I've already mentioned this on a different PR, so I'll just put a link in here to that comment rather than repeat myself: https://github.com/green-fox-academy/IceAge-Reddit-Backend/pull/6/files/07016b4dc5fc8be0aa06b384073c0130708fff28#r532276652

I've spent a couple minutes looking at this `validate` and `checkpassword`. To be fair, it is really complex and I'd be interested in either:
- your explanation of the code in a 1on1 session to defend your code
- possible changes of this implementation to make it more readable

Keep in mind that I am NOT saying this implementation is wrong OR this is horrible code. I do not understand these 6 lines from looking and them and I would have to go step through it to understand what is going on, that is why I want your explanation. It is also a sign of bad code, but it may not be the case.

Reply:
Code commented and also better renamed.